### PR TITLE
Fix feed source name editing

### PR DESCRIPTION
**Pull request by philip-cline**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JSDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

Currently, when editing the name of a feed source, beginning additions to the name with spaces will cause the spaces to be trimmed as soon as they are entered. This PR resolves that issue.

### CODE REVIEW

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
~This fix works but I'm not quite sure why we're making this change? Is there some use case for having a project name start with spaces? It seems like doing so will only cause confusion with no upsides.~

I was editing project titles, not feed source titles. I understand what's happening now. Sorry for the confusion!

File: lib/manager/components/GeneralSettings.js [hunk 1]
```diff
@@ -40,9 +40,11 @@ type State = {
 export default class GeneralSettings extends Component<Props, State> {
   state = {}
 
-  _onChange = ({target}: SyntheticInputEvent<HTMLInputElement>) => {
+  _onChange = ({target}: SyntheticInputEvent<HTMLInputElement>, key: 'name' | 'url') => {
```
Comment (miles-grant-ibigroup) [inline lib/manager/components/GeneralSettings.js]: You don't need a `key` variable, you should be able to use `target.name` to pull the value from the html tag

File: lib/manager/components/GeneralSettings.js [hunk 2]
```diff
@@ -40,9 +40,11 @@ type State = {
...
+    let value
     // Change empty string to null to avoid setting URL to empty string value.
-    const value = target.value === '' ? null : target.value.trim()
+    if (key === 'url') value = target.value === '' ? null : target.value.trim()
+    else value = target.value
```
Comment (miles-grant-ibigroup) [inline lib/manager/components/GeneralSettings.js]: Would it not be cleaner to always set `value = target.value` unless it's a url in which case set the trim inside an if block? That way it's one fewer check

File: lib/manager/components/GeneralSettings.js [hunk 3]
```diff
@@ -130,7 +132,7 @@ export default class GeneralSettings extends Component<Props, State> {
                     value={this._getFormValue('name')}
                     name={'name'}
                     disabled={disabled}
-                    onChange={this._onChange} />
+                    onChange={(e) => this._onChange(e, 'name')} />
```
Comment (miles-grant-ibigroup) [inline lib/manager/components/GeneralSettings.js]: This change can be reverted once we switch to using `target.name`

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
Fix works great, just a few opportunities for cleanup!

File: lib/manager/components/GeneralSettings.js (hunk 1 above)
Comment (philip-cline) [reply lib/manager/components/GeneralSettings.js]: Astounding. Changed in https://github.com/ibi-group/datatools-ui/pull/841/commits/07ed8e52df2b04bd228f7bf2d850f03a41005f9a

File: lib/manager/components/GeneralSettings.js (hunk 2 above)
Comment (philip-cline) [reply lib/manager/components/GeneralSettings.js]: Much cleaner. See https://github.com/ibi-group/datatools-ui/pull/841/commits/07ed8e52df2b04bd228f7bf2d850f03a41005f9a

File: lib/manager/components/GeneralSettings.js (hunk 3 above)
Comment (philip-cline) [reply lib/manager/components/GeneralSettings.js]: You betcha.

**Review by miles-grant-ibigroup — APPROVED** [review]
Wow this is much cleaner! Thanks for this

**Review by daniel-heppner-ibigroup — APPROVED** [review]
Seems like before spaces weren't allowed at all. This feels like a `feat: add supports for spaces`.

**Pull request merged**